apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -192,6 +200,30 @@ def get_query_server_config(name='beeswax', connector=None):
         else:
           cache.set("hiveserver2", json.dumps({"host": HIVE_SERVER_HOST.get(), "port": HIVE_HTTP_THRIFT_PORT.get()}))
 
+      # Replace ActiveEndpoint if the current HS2 is down
```

Comment:
Nice!

Just is there any way to avoid the duplication?
And add some tests https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/server/dbms_tests.py?

(in the past this was not done and this feature is extremely flaky)

Reply:
Started to work on this

Reply:
This is fixed locally.. and also fixed dbms_tests as well.